lib/devise/models/database_authenticatable.rb

```diff
@@ -41,7 +41,7 @@ def self.required_fields(klass)
       # For legacy reasons, we use `encrypted_password` to store
       # the hashed password.
       def password=(new_password)
-        attribute_will_change! 'password'
+        attribute_will_change! 'password' if defined? attribute_will_change!
```

Comment:
@twalpole Do we really need this `attribute_will_change! `? I've removed it and all tests is passing. I think we don't this track, what do you think?